src/ui/control/geolocate_control.js

```diff
@@ -476,6 +498,28 @@ class GeolocateControl extends Evented {
 
 export default GeolocateControl;
 
+function getDistance(latlng1, latlng2) {
```

Comment:
I don't think this needs to be addressed within this PR, but eventually we should eliminate the duplication between this and https://github.com/mapbox/mapbox-gl-js/blob/74b15736d414023345abab69b91995d949f7d11e/src/ui/control/scale_control.js#L124-L137 and in turn hopefully address #8777.

Reply:
Yeah agreed it's a copy/paste job -- good spot to refactor these into some helpers in the future.

Reply:
We'll likely forget about it later so let's eliminate duplication now — I think the best place to put it is `LngLat` `distanceTo(lngLat)` method. This would follow the same convention as Leaflet.

Reply:
Opened a separate PR for this: https://github.com/mapbox/mapbox-gl-js/pull/9202